lib/datadog/core/configuration/option.rb

```diff
@@ -94,6 +96,87 @@ def default_precedence?
 
         private
 
+        def coerce_env_variable(value)
+          case @definition.type
+          when :int
+            value.to_i
+          when :float
+            value.to_f
+          when :array
+            values = if value.include?(',')
+                       value.split(',')
+                     else
+                       value.split(' ') # rubocop:disable Style/RedundantArgument
+                     end
+
+            values.map! do |v|
+              v.gsub!(/\A[\s,]*|[\s,]*\Z/, '')
+
+              v.empty? ? nil : v
+            end
+
+            values.compact!
+            values
+          when :bool
+            string_value = value
+            string_value = string_value.downcase
+            string_value == 'true' || string_value == '1' # rubocop:disable Style/MultipleComparison
+          else
+            value
+          end
+        end
+
+        def validate_type(value)
+          raise_error = false
+
+          valid_type = validate(@definition.type, value)
+
+          unless valid_type
+            raise_error = if @definition.type_options[:nil]
```

Comment:
I might prefer `:nilable` (or `:allow_nil`) here.

Example:

```
o.type :float, nilable: true
```

Reply:
I'm ok with whatever the team decides. 
`nilable: true` sounds good

Reply:
👍 for `nilable`

Reply:
Done https://github.com/DataDog/dd-trace-rb/pull/2970/commits/345454af4bfc3e57b86157aa6d89f07892d24488